src/utils/SpokeUtils.ts

```diff
@@ -72,14 +53,36 @@ export async function getBlockRangeForDepositId(
     }
   });
 
+  // Define a mapping of block numbers to number of deposits at that block. This saves repeated lookups.
+  const queriedIds: Record<number, number> = {};
+
+  // Define a llambda function to get the deposit ID at a block number. This function will first check the
+  // queriedIds cache to see if the deposit ID at the block number has already been queried. If not, it will
+  // make an eth_call request to get the deposit ID at the block number. It will then cache the deposit ID
+  // in the queriedIds cache.
+  const _getDepositIdAtBlock = async (blockNumber: number): Promise<number> => {
+    if (queriedIds[blockNumber] === undefined) {
+      queriedIds[blockNumber] = await spokePool._getDepositIdAtBlock(blockNumber);
+    }
+    return queriedIds[blockNumber];
+  };
+
+  // Get the the deposit ID at the low block, and the deposit ID at the high block in parallel.
+  const [highestPossibleDepositIdInRange, lowestPossibleDepositIdInRange] = await Promise.all([
+    _getDepositIdAtBlock(initHigh),
+    _getDepositIdAtBlock(Math.max(deploymentBlock, initLow - 1)),
+  ]);
+
   // If the deposit ID at the initial high block is less than the target deposit ID, then we know that
   // the target deposit ID must be greater than the initial high block, so we can throw an error.
   if (highestPossibleDepositIdInRange <= targetDepositId) {
     // initLow   = 5: Deposits Num: 10
     //                                     // targetId = 11  <- fail (triggers this error)          // 10 <= 11
     //                                     // targetId = 10  <- fail (triggers this error)          // 10 <= 10
     //                                     // targetId = 09  <- pass (does not trigger this error)  // 10 <= 09
-    throw new Error(`Target depositId is greater than the initial high block (${targetDepositId} > ${initHigh})`);
+    throw new Error(
+      `Target depositId is greater than the initial high block (${targetDepositId} > ${highestPossibleDepositIdInRange})`
```

Comment:
`highestPossibleDepositIdInRange` pushes us out over 120 chars here, and prettier unfortunately doesn't propose a refactor to fix that. Would you consider `highestDepositIdInRange` instead?

The error `Target depositId out of range (${targetDepositId} > ${highestDepositIdInRange})` might also help to consolidate it a bit.